theseus/geometry/functional/lie_group.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+import abc
+
+from typing import List, Optional
+from .utils import check_jacobians_list
+
+# There are four functions associated with each Lie group operator xxx.
+# _xxx_impl: mathematical implementation of the operator
+# _j_xxx_impl: mathematical implementation of the operator jacobian
+# _xxx_base: a torch.autograd.Function wrapper of _xxx_impl
+# _j_xxx_base: simply equivalent to _j_xxx_impl
```

Comment:
wondering if `_xxx_wrapper` is a better name since `base` makes me think of the low level math which is actually in `impl`.

Also the comment for `_j_xxx_base` should also include torch.autograd.Function in the text similar to `_xxx_base`.

Reply:
The implementation of `torch.autograd.Function` for `_j_xxx_impl` is planned for 2023 H1.

Reply:
Based on offline discussion here are functions we need

```python
_exp_impl  # analytic op
_jexp_impl  # analytic jop
_exp_wrapper  # torch autograd op
_jexp_wrapper = _jexp_impl # later torch autograd jop

exp = _exp_wrapper
jexp = _jexp_wrapper
```